var/spack/repos/builtin/packages/gaudi/package.py

```diff
@@ -11,41 +11,49 @@ class Gaudi(CMakePackage):
 
     homepage = "http://gaudi.web.cern.ch/gaudi/"
     git      = "https://gitlab.cern.ch/gaudi/Gaudi.git"
+    url      = "https://gitlab.cern.ch/gaudi/Gaudi/-/archive/v33r1/Gaudi-v33r1.tar.gz"
 
-    version('develop', branch='master')
-    version('30.5',    commit='2c70e73ee5b543b26197b90dd59ea4e4d359d230')
+    version('master', branch='master')
+    # major cmake config overhaul already in use by some
+    version('develop', git='https://gitlab.cern.ch/clemenci/Gaudi.git', branch='cmake-modernisation')
+    version('33.1', sha256='7eb6b2af64aeb965228d4b6ea66c7f9f57f832f93d5b8ad55c9105235af5b042')
+    version('33.0', sha256='76a967c41f579acc432593d498875dd4dc1f8afd5061e692741a355a9cf233c8')
+    version('32.2', sha256='e9ef3eb57fd9ac7b9d5647e278a84b2e6263f29f0b14dbe1321667d44d969d2e')
     version('31.0',    commit='aeb156f0c40571b5753a9e1dab31e331491b2f3e')
+    version('30.5',    commit='2c70e73ee5b543b26197b90dd59ea4e4d359d230')
+
+    maintainers = ['drbenmorgan', "vvolkl"]
 
-    variant('tests', default=False,
-            description='Prepare to run the test suite')
     variant('optional', default=False,
-            description='Build most optional components')
+            description='Build most optional components and tests, including GaudiExamples')
     variant('vtune', default=False,
             description='Build with Intel VTune profiler support')
 
+    # only build subdirectory GaudiExamples when +optional
+    patch("build_testing.patch", when="@:33.1")
+    # fix for the new cmake config, should be merged in branch
+    patch('python2.patch', when="@develop")
+    # fixes for the cmake config which could not find newer boost versions
+    patch("link_target_fixes.patch", when="@33.0:33.1")
+    patch("link_target_fixes32.patch", when="@:32.2")
+
     # These dependencies are needed for a minimal Gaudi build
     depends_on('boost@1.67.0: +python')
     depends_on('cmake', type='build')
+    depends_on('cppgsl')
     depends_on('intel-tbb')
     depends_on('libuuid')
-    depends_on('python@:2.99.99')
-    depends_on('py-xenv@develop_2018-12-20:')
+    # some bugs with python 3.8
+    depends_on('python@:3.7.99', type=('build', 'run'))
```

Comment:
I assume versions 30.5 and 31.0 still require Python 2?

Reply:
Indeed, good catch, added the correct dependencies.